cli/src/pcluster/models/login_nodes_status.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from enum import Enum
+
+from pcluster.aws.aws_api import AWSApi
+from pcluster.utils import get_chunks
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LoginNodesPoolState(Enum):
+    """Represents the internal status of the login nodes pools."""
+
+    PENDING = "pending"
+    ACTIVE = "active"
+    FAILED = "failed"
+
+    def __str__(self):
+        return str(self.value)
+
+
+class LoginNodesStatus:
+    """Represents the status of the cluster login nodes pools."""
+
+    def __init__(self, stack_name):
+        self.stack_name = stack_name
+        self.login_nodes_pool_available = False
+        self.load_balancer_arn = None
+        self.target_group_arn = None
+        self.status = None
+        self.dns_name = None
+        self.scheme = None
+        self.healthy_nodes = None
+        self.unhealthy_nodes = None
```

Comment:
[Nitpick] Since you've defined the getters, you can make these fields private prepending them with an `_` and then mark the getter function as `@Property` in order to simplify their usage.

Reply:
Marked them as private.